servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultHttpLoadBalancerFactory.java

```diff
@@ -214,4 +278,188 @@ public String toString() {
+        private final Function<Throwable, ErrorClass> errorClassFunction;
+        private final Function<HttpResponseMetaData, ErrorClass> peerResponseErrorClassifier;
+        @Nullable
+        private final RequestTracker tracker;
+
+        DefaultHttpLoadBalancedConnection(final FilterableStreamingHttpConnection delegate,
+                                          final ReservableRequestConcurrencyController concurrencyController,
+                                          final Function<Throwable, ErrorClass> errorClassFunction,
+                                          final Function<HttpResponseMetaData, ErrorClass> peerResponseErrorClassifier,
+                                          @Nullable final RequestTracker tracker) {
+            this.delegate = delegate;
+            this.concurrencyController = concurrencyController;
+            this.errorClassFunction = errorClassFunction;
+            this.peerResponseErrorClassifier = peerResponseErrorClassifier;
+            this.tracker = tracker;
+        }
+
+        @Override
+        public int score() {
+            return 1;
+        }
+
+        @Override
+        public ReservedHttpConnection asConnection() {
+            return toReservedConnection(this, executionContext().executionStrategy());
+        }
+
+        @Override
+        public ReservedBlockingStreamingHttpConnection asBlockingStreamingConnection() {
+            return toReservedBlockingStreamingConnection(this, executionContext().executionStrategy());
+        }
+
+        @Override
+        public ReservedBlockingHttpConnection asBlockingConnection() {
+            return toReservedBlockingConnection(this, executionContext().executionStrategy());
+        }
+
+        @Override
+        public Completable releaseAsync() {
+            return concurrencyController.releaseAsync();
+        }
+
+        @Override
+        public Completable closeAsyncGracefully() {
+            return delegate.closeAsyncGracefully();
+        }
+
+        @Override
+        public Result tryRequest() {
+            return concurrencyController.tryRequest();
+        }
+
+        @Override
+        public boolean tryReserve() {
+            return concurrencyController.tryReserve();
+        }
+
+        @Override
+        public void requestFinished() {
+            concurrencyController.requestFinished();
+        }
+
+        @Override
+        public HttpConnectionContext connectionContext() {
+            return delegate.connectionContext();
+        }
+
+        @Override
+        public <T> Publisher<? extends T> transportEventStream(final HttpEventKey<T> eventKey) {
+            return delegate.transportEventStream(eventKey);
+        }
+
+        @Override
+        public Single<StreamingHttpResponse> request(final StreamingHttpRequest request) {
+            return Single.defer(() -> {
+                if (tracker == null) {
+                    return delegate.request(request).shareContextOnSubscribe();
```

Comment:
bcz the presence of tracker is known upfront and never changes, consider adding this check before using `Single.defer`